core/src/main/scala/com/permutive/refreshable/Refreshable.scala

```diff
@@ -26,9 +26,7 @@ import retry._
 
 import scala.concurrent.duration._
 
-trait Refreshable[F[_], A] { self =>
-
-  implicit protected def functor: Functor[F]
+abstract class Refreshable[F[_]: Functor, A] { self =>
```

Comment:
@TimWSpence should we make this `sealed`? It's extended in `prometheus4cats-contrib` library [here](https://github.com/permutive-engineering/prometheus4cats-contrib/blob/main/refreshable/src/main/scala/prometheus4cats/refreshable/InstrumentedRefreshable.scala#L34)